File: src/main/java/tk/skulk/plugin/core/extensions/entityoverride/listeners/CreeperNoGriefListener.java
package tk.skulk.plugin.core.extensions.entityoverride.listeners;

import org.bukkit.entity.EntityType;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.entity.EntityExplodeEvent;
import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.extensions.entityoverride.EntityOverrideExtension;
import tk.skulk.plugin.wrapper.BaseListener;

public final class CreeperNoGriefListener extends BaseListener<EntityOverrideExtension> {
    public CreeperNoGriefListener(final @NotNull EntityOverrideExtension extension) {
        super(extension);
    }

    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    public void onEntityExplode(final @NotNull EntityExplodeEvent event) {
        if (event.getEntityType() != EntityType.CREEPER) {
            return;
        }

        event.setCancelled(true);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/entityOverride/listeners/CreeperNoGriefListener.kt
package tk.skulk.plugin.core.extensions.entityOverride.listeners

import org.bukkit.entity.EntityType
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.entity.EntityExplodeEvent
import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.entityOverride.EntityOverrideExtension
import tk.skulk.plugin.framework.Listener

class CreeperNoGriefListener(extension: EntityOverrideExtension) : Listener<SNPlugin, EntityOverrideExtension>(
    extension
) {
    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
    fun onEntityExplode(event: EntityExplodeEvent) {
        if (event.entityType != EntityType.CREEPER) {
            return
        }

        event.blockList().clear()
    }
}
